### PR TITLE
:sparkles: add private chat design

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+    match /config/{configFile} {
         allow write: if request.auth.uid != null;
         allow read: if true;
     }
-    match /{document=**} {
+    match /users/{userId} {
         allow write, read: if request.auth.uid != null;
     }
+    match /chatsv2/{chatId} {
+    		allow write: if request.auth.uid != null && request.resource.data.senderId == request.auth.uid;
```
Comment (mathieud) [inline firestore.rules:12]: Indentation seems off

File: firestore.rules [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+        allow read: if request.auth.uid != null && (resource.data.recipientId == null || resource.data.recipientId == request.auth.uid);
```
Comment (mathieud) [inline firestore.rules:13]: Does this allow you to read to messages you sent (where you are senderId)?

File: src/components/organisms/Chatbox/Chatbox.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  } else {
+    chatsToDisplay = chatsToDisplay.filter(
+      (chat: any) =>
+        !chat.recipientId ||
+        chat.recipientId === user.uid ||
+        chat.senderId === user.uid
+    );
+  }
+
+  console.log(chats.length);
```
Comment (mathieud) [inline src/components/organisms/Chatbox/Chatbox.tsx:58]: 🍩

File: firestore.rules (hunk 2 above)
Comment (antoineeudes) [reply firestore.rules:13]: that's true there 's a mistake

**Pull request merged**